templates/bake/Template/add.twig

```diff
@@ -17,6 +17,19 @@
 /**
  * @var \{{ namespace }}\View\AppView $this
  * @var \{{ entityClass }} ${{ singularVar }}
+        {{- "\n" }}
+{%- if associations.BelongsTo is defined %}
+    {%- for assocName, assocData in associations.BelongsTo %}
+ * @var array ${{ assocData.variable }}
```

Comment:
Lies!! This var is a result of `find('list')` which is a `Query` instance, not an array. So the type should be `iterable`.

Reply:
The concrete type isnt that important usually just the existence:)
Could usually also be string[] etc
As those are used for options.

Reply:
The concrete type isnt that important usually just the existence:)
Could usually also be string[] etc
As those are used for options.

Reply:
> The concrete type isnt that important...

I disagree, incorrect types are misleading and cause confusion for new users.

Reply:
@ADmad : I was thinking about this when adding the variable. On the index templates, you have this docblock:
```php
/**
 * @var \App\View\AppView $this
 * @var \App\Model\Entity\File[]|\Cake\Collection\CollectionInterface $files
 */
```

We could do something similar with add/edit.

```php
/**
 * @var \App\View\AppView $this
 * @var string[]|\Cake\Collection\CollectionInterface $files
 */
```

What do you think? I can create a PR for this if you'd like.

Reply:
`\Cake\Collection\CollectionInterface<string>` would be the most accurate but if that's not supported by IDEs(PhpStorm) yet, then `string[]|\Cake\Collection\CollectionInterface` is fine.

Reply:
Please use `\Cake\Collection\CollectionInterface|array<string>` or `\Cake\Collection\CollectionInterface|string[]` to follow our new standard.

Reply:
Even if it was, wouldn't it be a good idea to follow the conventions in other templates until all templates are changed?

I couldn't get it to work with PhpStorm 2021.1.4 at least.

While `\Cake\Collection\CollectionInterface<string>` doesn't throw any errors/warnings, say we tried `\Cake\Collection\CollectionInterface<\Cake\Datasource\EntityInterface>` for index it wouldn't suggest any properties/methods in `EntityInterface` when looping through it. So it doesn't seem to be fully supported yet.

I say we go with the `string[]|\Cake\Collection\CollectionInterface` way.